core/src/main/java/org/ehcache/core/spi/store/AbstractWrapperStoreProvider.java

```diff
@@ -16,18 +16,21 @@
 package org.ehcache.core.spi.store;
 
 import org.ehcache.core.collections.ConcurrentWeakIdentityHashMap;
+import org.ehcache.core.spi.service.StatisticsService;
 import org.ehcache.spi.service.Service;
 import org.ehcache.spi.service.ServiceConfiguration;
+import org.ehcache.spi.service.ServiceDependencies;
 import org.ehcache.spi.service.ServiceProvider;
 
 import java.util.Arrays;
 import java.util.Map;
 
 import static org.ehcache.core.store.StoreSupport.selectStoreProvider;
 
+@ServiceDependencies({StatisticsService.class})
 public abstract class AbstractWrapperStoreProvider implements WrapperStore.Provider {
 
-  private volatile ServiceProvider<Service> serviceProvider;
+  protected volatile ServiceProvider<Service> serviceProvider;
```

Comment:
If we do the stats context binding in this class we can put this back to being private (I think).